File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    result = []
    
    for query in queries:
        s, e, k = query
        sub_array = arr[s:e+1]  # 범위 내의 부분 배열 생성
        filtered_array = [x for x in sub_array if x > k]  # 조건을 만족하는 값들로 필터링
        
        if filtered_array:
            result.append(min(filtered_array))  # 필터링된 값 중 최솟값 추가
        else:
            result.append(-1)  # 조건을 만족하는 값이 없을 경우 -1 추가
    
    return result

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
    result = [];
    
    for (i = 0; i < queries.length; i++) {
        query = queries[i];
        s = query[0];
        e = query[1];
        k = query[2];
        sub = arr.slice(s, e + 1);
        
        filter = [];
        for (let j = 0; j < sub.length; j++) {
            if (sub[j] > k) {
                filter.push(sub[j]);
            }
        }
        
        if (filter.length > 0) {
            min = filter[0];
            for (j = 1; j < filter.length; j++) {
                if (filter[j] < min) {
                    min = filter[j];
                }
            }
        result.push(min);
        } else {
            result.push(-1);
        }
    }
    return result;
}